src/LaravelNomadServiceProvider.php

```diff
@@ -23,9 +28,17 @@ public function boot()
      */
     public function register()
     {
-        $this->app->bind('db.connection.mysql', 'ShiftOneLabs\LaravelNomad\Extension\Database\MySqlConnection');
-        $this->app->bind('db.connection.pgsql', 'ShiftOneLabs\LaravelNomad\Extension\Database\PostgresConnection');
-        $this->app->bind('db.connection.sqlite', 'ShiftOneLabs\LaravelNomad\Extension\Database\SQLiteConnection');
-        $this->app->bind('db.connection.sqlsrv', 'ShiftOneLabs\LaravelNomad\Extension\Database\SqlServerConnection');
+        Connection::resolverFor('mysql', function ($connection, $database, $prefix, $config) {
+            return new MySqlConnection($connection, $database, $prefix, $config);
+        });
+        Connection::resolverFor('pgsql', function ($connection, $database, $prefix, $config) {
+            return new PostgresConnection($connection, $database, $prefix, $config);
+        });
+        Connection::resolverFor('sqlite', function ($connection, $database, $prefix, $config) {
+            return new SQLiteConnection($connection, $database, $prefix, $config);
+        });
+        Connection::resolverFor('sqlsrv', function ($connection, $database, $prefix, $config) {
+            return new SqlServerConnection($connection, $database, $prefix, $config);
+        });
```

Comment:
Do a check on `method_exists(Connection::class, 'resolverFor')`. If the method exists, use this new registration method. If the method doesn't exist, keep the old bindings.